src/Symfony/Component/Security/Core/Authorization/TraceableAccessDecisionManager.php

```diff
@@ -51,10 +52,25 @@ public function decide(TokenInterface $token, array $attributes, $object = null)
     {
         $result = $this->manager->decide($token, $attributes, $object);
 
+        $voterDetails = array();
+
+        foreach ($this->voters as $voter) {
+            if ($voter instanceof TraceableVoter) {
+                $voterClass = $voter->getVoterClass();
+                $voteLog = $voter->getVoteLog();
```

Comment:
This implementation won't work fine in many cases:

- if a voter is not called at all for this decision (only the `consensus` strategy is guaranteed to always call all voters. `affirmative` has a shortcut when granting access, and `unanimous` has one when denying access), you will retrieve the log of a previous decision
- if a voter triggers another decision internally, the log being retrieved here will be the wrong one

Reply:
@stof thank you for your review, I'm going to think again about my implementation. However one of your comments is not very clear for me:
> **if a voter triggers another decision internally**, the log being retrieved here will be the wrong one

How is it possible ?

Reply:
Well, you can inject the AccessDecisionManager inside your voter (we made that possible in 3.3 or 3.4, I dont remember)

Reply:
Hum indeed... So in this case, the profiler displays the decision of the ADM  called by the voter with the other decisions. This behavior seems unexpected to me. Do you think it's a correct behavior or does it deserve to be fixed ?

Reply:
I created an other PR for that: #28855